src/providers/details/tabs/Hosts/components/VSphereNetworkModal.tsx

```diff
@@ -0,0 +1,101 @@
+import { type FC, type ReactNode, useCallback, useState } from 'react';
+import { validateNoSpaces } from 'src/modules/Providers/utils/validators/common';
+import { useForkliftTranslation } from 'src/utils/i18n';
+
+import ModalForm from '@components/ModalForm/ModalForm';
+import type { NetworkAdapters, V1beta1Provider } from '@kubev2v/types';
+import { Form, ModalVariant } from '@patternfly/react-core';
+
+import { VSphereEndpointType } from './utils/constants';
+import { getSelectedInventoryHostNetworkTriples } from './utils/getSelectedInventoryHostNetworkTriples';
+import { onSaveHost } from './utils/onSaveHost';
+import type { InventoryHostNetworkTriple } from './utils/types';
+import HostsNetworksSelect from './HostsNetworksSelect';
+import HostsNetworksSetPassword from './HostsNetworksSetPassword';
+import HostsNetworksSetUserName from './HostsNetworksSetUserName';
+
+import './VSphereNetworkModal.style.css';
+
+type VSphereNetworkModalProps = {
+  provider: V1beta1Provider;
+  data: InventoryHostNetworkTriple[];
+  selectedIds: string[];
+};
+
+const VSphereNetworkModal: FC<VSphereNetworkModalProps> = ({ data, provider, selectedIds }) => {
+  const { t } = useForkliftTranslation();
+  const [alertMessage, setAlertMessage] = useState<ReactNode>(null);
+  const [network, setNetwork] = useState<NetworkAdapters | undefined>(undefined);
+  const [username, setUsername] = useState<string | undefined>(undefined);
+  const [password, setPassword] = useState<string | undefined>(undefined);
+
+  const selectedLength = selectedIds.length;
+  const endpointType = provider?.spec?.settings?.sdkEndpoint ?? '';
+
+  const shouldDisableSave = () => {
+    if (endpointType === VSphereEndpointType.ESXi.valueOf()) {
+      return network === undefined;
+    }
+
+    return (
+      network === undefined ||
+      !username ||
+      !password ||
+      !validateNoSpaces(username) ||
+      !validateNoSpaces(password)
+    );
+  };
+
+  const selectedInventoryHostPairs = getSelectedInventoryHostNetworkTriples(data, selectedIds);
+
+  const handleSave = useCallback(async () => {
+    setAlertMessage(null);
```

Comment:
I don't see setAlertMessage, used or passed anywhere except here, its default value is null, and u set it to null.. why? If u don't need it, just delete it